tools/Modules/Tests/RepoModuleTests.ps1

```diff
@@ -8,3 +8,254 @@
     Set-SPNRole -ADAppDisplayName "MyModTestApp1" -SubscriptionId 2c224e7e-3ef5-431d-a57b-e71f4662e3a6
     Remove-ServicePrincipal -ADAppDisplayName "MyModTestApp1" -SubscriptionId 2c224e7e-3ef5-431d-a57b-e71f4662e3a6 -TenantId 72f988bf-86f1-41af-91ab-2d7cd011db47
 }
+
+
+Function Test-NewCredentialNewServicePrincipal
```

Comment:
Is the idea to run these whenever we update the modules?  After this is merged we should (1) adapt these to be Pester tests; (2) Make sure these are run as part of ci when appropriate

Reply:
I'll add that to the issue and start working on it as soon as this PR is merged.